### PR TITLE
fix: Last navigation dropdown is out of the window #2860

### LINKED ISSUE #2860: Last navigation dropdown is out of the window
In dev17, on medium size screen:

Open the `Accessories` navigation dropdown. Then the dropdown content is out of the window:

obraz

### PR DESCRIPTION
**Pull request by Pucek9**
Closes #2860

### CODE REVIEW

**Comment by Pucek9** [comment]
tests will be done after first review

File: projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts [hunk 1]
```diff
@@ -108,4 +110,43 @@ export class NavigationUIComponent implements OnDestroy {
...
+        this.renderer.setStyle(
+          wrapper,
+          'margin-left',
+          `${node.offsetWidth - wrapper.offsetWidth}px`
+        );
+      }
+    }
+  }
+
+  private validateWrappersAndAlignToRightIfStickOut() {
```
Comment (kacperknapik) [inline projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts]: maybe we should think about shortening this name - `validateWrappersAndAlignToRightIfStickOut()`

File: projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts [hunk 2]
```diff
@@ -108,4 +110,43 @@ export class NavigationUIComponent implements OnDestroy {
       this.subscription.unsubscribe();
     }
   }
+
+  private alignWrapperToRightIfStickOut(node: HTMLElement) {
```
Comment (kacperknapik) [inline projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts]: maybe we should think about shortening this name - `alignWrapperToRightIfStickOut()`

File: projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts [hunk 3]
```diff
@@ -47,18 +49,28 @@ export class NavigationUIComponent implements OnDestroy {
 
   private subscription: Subscription;
 
-  constructor(private router: Router, private renderer: Renderer2) {
+  @HostListener('window:resize')
+  onResize() {
```
Comment (kacperknapik) [inline projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts]: shouldn't we add some kind of debouncer here? performance might be affected by many resizes

Comment (Pucek9) [reply projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts]: that's nice idea

File: projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts (hunk 1 above)
Comment (Pucek9) [reply projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts]: changed to alignWrappersToRightIfStickOut

File: projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts (hunk 2 above)
Comment (Pucek9) [reply projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts]: I don't have a better idea for a name that describes what this function does

**Pull request merged**